internal/vds/openvdsvoxelvector.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef VDS_SLICE_OPENVDSVOXELVECTOR_HPP
+#define VDS_SLICE_OPENVDSVOXELVECTOR_HPP
+
+#include <string>
+#include <stdexcept>
+
+#include <OpenVDS/Vector.h>
+
+#include "metadatahandle.hpp"
+
+/**
+ * Represents a vector in a coordinate system used by openvds Request- functions.
+
+     |---Voxel index 0---|----Voxel index 1---|........|-----Voxel index nsamples-1------|
+     |                   |                    |        |                                 |
+     |                   |                    |        |                                 |
+ -> (0)-----(0.5)-------(1)-------(1.5)------(2)...(nsamples-1)---(nsamples-0.5)------(nsamples)
+              |                     |                                  |
+              |                     |                                  |
+              |                     |                                  |
+     Voxel position 0       Voxel position 1               Voxel position nsamples-1
+          Sample 0               Sample 1                     Sample nsamples-1
+
+ * -> Shows how coordinate is stored in OpenvdsVoxelVector and its relation to actual
+ * sample positions and results from ToVoxelIndex/ToVoxelPosition functions.
+ *
+ * In contrast to openvds data in the vector are always stored in
+ * Inline/Crossline/Sample order.
+ */
+class OpenvdsVoxelVector : public OpenVDS::DoubleVector3{
```

Comment:
I'm not super chuffed about this class. I guess I fail to see the justification for adding a wrapper around the point class. Probably because for me this entire endouvior is about the mismatch between the output of OpenVDS's transformers and what it expects in its access manager. Hence, what makes sense in my head is to make a wrapper around the open-vds transformer so that we can add the 0.5 offset. But here we have 2 wrappers for one thing and I don't follow why we want that?

Reply:
> One of these issues is a DoubleVector. It is just a data array, with no meaning behind it. To know what it contains and be able to use it correctly you need to know which function created that double vector. Was that to IJKIndex or was that to VoxelPosition? In theory a separate class prevents you from using any DoubleVector in a place where I actually want only those "Double Vectors which represent coordinates to Request functions".

Sounds like you want a simple typedef maybe?

> I think this is a quite frequent problem: should this datastructure that I use stay raw, or should I add a light wrapper over it for convenience?

True and to answer that we weigh up the pros vs cons. I.e. added complexity vs convenience. So then my question is, what convenience does it add? We can now do
`voxel.InlineValue()` instead of `voxel[inline.dimension()]`.  Which maybe looks a bit nicer, but introduces tighter coupling. Personally I think the complexity over adding a new class and having to understand how it related to openvds out weighs the benefit here. 

But consider this as questions more than request for change